### PR TITLE
Fixes #513 - AdvancedCollectionView not working with AoT

### LINKED ISSUE #513: AdvancedCollectionView items are empty in gallery sample under Windows App SDK
### Describe the bug

When running the gallery with the Wasdk head on the latest `main` commit, the 'AdvancedCollectionView' sample does not display items properly. The items are in the list, but the contents are empty. 

Wasdk:
image

UWP:
image

### Steps to reproduce

```text
1. Clone the repo
2. Generate the gallery solution, include the Collections component and the wasdk multitarget/head.
3. Build and deploy the wasdk gallery 
4. Navigate to the 'AdvancedCollectionView' sample page and observe empty list items.
```

### Expected behavior

Contents should render the same as they did before Wasdk 1.6 and the same as on UWP. 

### Screenshots

_No response_

### Code Platform

- [ ] UWP
- [X] WinAppSDK / WinUI 3
- [ ] Web Assembly (WASM)
- [ ] Android
- [ ] iOS
- [ ] MacOS
- [ ] Linux / GTK

### Windows Build Number

- [ ] Windows 10 1809 (Build 17763)
- [ ] Windows 10 1903 (Build 18362)
- [ ] Windows 10 1909 (Build 18363)
- [ ] Windows 10 2004 (Build 19041)
- [ ] Windows 10 20H2 (Build 19042)
- [ ] Windows 10 21H1 (Build 19043)
- [ ] Windows 10 21H2 (Build 19044)
- [ ] Windows 10 22H2 (Build 19045)
- [ ] Windows 11 21H2 (Build 22000)
- [ ] Other (specify)

### Other Windows Build number

_No response_

### App minimum and target SDK version

- [ ] Windows 10, version 1809 (Build 17763)
- [ ] Windows 10, version 1903 (Build 18362)
- [ ] Windows 10, version 1909 (Build 18363)
- [ ] Windows 10, version 2004 (Build 19041)
- [ ] Windows 10, version 2104 (Build 20348)
- [ ] Windows 11, version 22H2 (Build 22000)
- [ ] Other (specify)

### Other SDK version

_No response_

### Visual Studio Version

_No response_

### Visual Studio Build Number

_No response_

### Device form factor

_No response_

### Additional context

_No response_

### Help us help you

Yes, I'd like to be assigned to work on this item.

### PR DESCRIPTION
**Pull request by michael-hawker**
## Fixes #513

Similar fix to #514 along with a bit of extra clean-up to simplify samples.

## PR Type

What kind of change does this PR introduce?

Mainly using x:Bind instead of Binding to make sample work with AoT.

## What is the current behavior?

Doesn't display information in list.

## What is the new behavior?

Information displayed.

## PR Checklist

Please check if your PR fulfills the following requirements: 

- [ ] Created a feature/dev branch in your fork (vs. submitting directly from a commit on main)
- [ ] Based off latest main branch of toolkit
- [ ] Tested code with current supported SDKs
- [ ] New component
  - [ ] Documentation has been added
  - [ ] Sample in sample app has been added
  - [ ] Analyzers are passing for documentation and samples
  - [ ] Icon has been created (if new sample) following the [Thumbnail Style Guide and templates](https://github.com/CommunityToolkit/WindowsCommunityToolkit-design-assets)
- [ ] Tests for the changes have been added (if applicable)
- [ ] Header has been added to all new source files
- [ ] Contains **NO** breaking changes

## Other information

Tested in Debug/Release

### CODE REVIEW

File: components/Triggers/src/CompareStateTrigger.cs [hunk 1]
```diff
@@ -13,7 +13,7 @@ namespace CommunityToolkit.WinUI;
     /// <para>
     /// Example: Trigger if a value is greater than 0
     /// <code lang="xaml">
-    ///     &lt;triggers:CompareStateTrigger Value="{Binding MyValue}" CompareTo="0" Comparison="GreaterThan" />
+    ///     &lt;triggers:CompareStateTrigger Value="{x:Bind MyValue}" CompareTo="0" Comparison="GreaterThan" />
```
Comment (Jay-o-Way) [inline components/Triggers/src/CompareStateTrigger.cs:16]: Don't know how this bit is used, but... Are you aware that the default mode of `x:Bind` is **OneTime**, in stead of OneWay for Binding?

Comment (michael-hawker) [reply components/Triggers/src/CompareStateTrigger.cs:16]: Good catch @Jay-o-Way, yes that's true. I just did a mindless replace here. This is just the doc comment example, but we should ensure its proper.

Comment (michael-hawker) [inline components/Triggers/src/CompareStateTrigger.cs:16]:
```suggestion
    ///     &lt;triggers:CompareStateTrigger Value="{x:Bind MyValue, Mode=OneWay}" CompareTo="0" Comparison="GreaterThan" />
```

File: components/Triggers/src/IsEqualStateTrigger.cs [hunk 2]
```diff
@@ -13,7 +13,7 @@ namespace CommunityToolkit.WinUI;
 /// <para>
 /// Example: Trigger if a value is null
 /// <code lang="xaml">
-///     &lt;triggers:EqualsStateTrigger Value="{Binding MyObject}" EqualTo="{x:Null}" />
+///     &lt;triggers:EqualsStateTrigger Value="{x:Bind MyObject}" EqualTo="{x:Null}" />
```
Comment (michael-hawker) [inline components/Triggers/src/IsEqualStateTrigger.cs:16]:
```suggestion
///     &lt;triggers:EqualsStateTrigger Value="{x:Bind MyObject, Mode=OneWay}" EqualTo="{x:Null}" />
```

**Review by michael-hawker — COMMENTED** [review]
Adding suggestions from @Jay-o-Way

**Review by Arlodotexe — APPROVED** [review]
Tested locally, looks good now!

image

**Pull request merged**